core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -1,15 +1,169 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { DataSource } from '@angular/cdk/table';
+import { MatPaginator, MatTableDataSource } from '@angular/material';
 
+
+import { ExecuteWorkflowService } from './../../service/execute-workflow/execute-workflow.service';
+
+import { Observable } from 'rxjs/Observable';
+import { NgbModal , ModalDismissReasons } from '@ng-bootstrap/ng-bootstrap';
+import { ExecutionResult, SuccessExecutionResult } from './../../types/workflow-execute.interface';
+
+/**
+ * ResultPanelCompoent is the bottom level area that
+ *  displays the execution result of a workflow after
+ *  the execution finishes.
+ *
+ * The Component will display the result in an excel
+ *  table format, where each row represents a result
+ *  from the workflow and each column represents the
+ *  type of result the workflow returns.
+ *
+ * Clicking each row of the result table will create an
+ *  pop-up window and display the detail of that row
+ *  in a pretty json format.
+ *
+ * @author Henry Chen
+ * @author Zuozhi Wang
+ *
+ */
 @Component({
   selector: 'texera-result-panel',
   templateUrl: './result-panel.component.html',
   styleUrls: ['./result-panel.component.scss']
 })
 export class ResultPanelComponent implements OnInit {
 
-  constructor() { }
+  public showMessage: boolean = false;
+  public message: string = '';
+
+  public currentColumns: TableColumn[] | undefined;
+  public currentDisplayColumns: string[] | undefined;
+  public currentDataSource: MatTableDataSource<object> | undefined;
+  public currentDisplayRow: string = '';
+
+  @ViewChild(MatPaginator) paginator: MatPaginator | null = null;
+
+  constructor(private executeWorkflowService: ExecuteWorkflowService, private modalService: NgbModal) {
+    // once an execution has ended, update the result panel to dispaly
+    //  execution result or error
+    this.executeWorkflowService.getExecuteEndedStream().subscribe(
+      executionResult => this.handleResultData(executionResult),
+    );
+  }
+
+  /**
+   * Opens the ng-bootstrap model to display the row details in
+   *  pretty json format when clicked. User can view the details
+   *  in a larger, expanded format.
+   *
+   * @param content
+   */
+  public open(content: any): void {
+    this.modalService.open(content).result.then((result) => {
```

Comment:
How can this function accept an argument which is in constructor but wasn't assigned to any local data member?

Reply:
`content` is a `TemplateRef` defined by the line `<ng-template #content let-c="close" let-d="dismiss">`, as described in the ng-bootstrap library: https://ng-bootstrap.github.io/#/components/modal/api.